### PR TITLE
UP-4978: Apply xmlns attribute globally to stylesheet

### PR DESCRIPTION
**Pull request by mairi**
This change prevents import errors and test failures caused by empty xmlns attributes appearing when a portlet definition is transformed by the data upgrade process.

https://issues.jasig.org/browse/UP-4978

##### Checklist

-   [x ] the [individual contributor license agreement][] is signed
-   [x ] commit message follows [commit guidelines][]

##### Description of change
In some circumstances when a portlet definition is passed through the automatic upgrade process, the lifecycle element in the resulting portlet definition has an empty xmlns attribute. This can cause:

- a test failure in testUpgradeTestPortlet43to50, as described on the issue tracker for UP-4978; 
- failures on the dataInit task in uPortal-start if local data includes portlet definitions that contain a lifecycle element.

From investigation, it looks as though the issue happens on the transform of the portlet definition in org.apereo.portal.io.xml.XsltDataUpgrader.java, similar to the problem described here: https://stackoverflow.com/questions/3504672/xslt-blank-xmlns-after-transform. 

I have tracked the cause down to the appearance of a literal element <lifecycle> in the file uPortal-webapp/src/main/resources/org/apereo/portal/io/xml/portlet/upgradePortlet_43.xsl.  I've implemented one of the suggestions from stack overflow - applying the xmlns attribute at stylesheet level. There is more than one way of fixing the problem, but this seemed like the simplest given the limited context in which it arises.

After applying the change locally:

- I no longer see the test failure in testUpgradeTestPortlet43to50; 
- I can successfully run dataInit within uPortal-start using my local uPortal UP-4978 branch without error.

This is my first pull request for uPortal. I have completed an ICLA but I'm not sure if there is (or needs to be) a CCLA for my organisation - The University of Edinburgh.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit

### CODE REVIEW

**Comment by bjagg** [comment]
No, A CCLA is not required by your institution. Thank you for the ICLA submission!

**Comment by apetro** [comment]
Hi @mairi 

Edinburgh [does not appear to have a CCLA on file with Apereo](http://licensing.apereo.org/).

Only you can say whether you need a CCLA from Edinburgh. ["Individuals are responsible for making sure their organization has completed a CCLA if their contributions are owned by their employer."](https://www.apereo.org/licensing). Apereo (and your fellow uPortal project participants) will rely upon your ICLA's representation that 

> 4. You represent that you are legally entitled to grant the above license. If your employer(s) has rights to intellectual property that you create that includes your Contributions, you represent that you have received permission to make Contributions on behalf of that employer, that your employer has waived such rights for your Contributions to the Foundation, or that your employer has executed a separate Corporate CLA with the Foundation.

Simply put: if you are able to contribute your contributions, awesome! If you're not able to contribute your contributions (e.g., because they're not yours and you can't get permission from the relevant parties), much less awesome.

I'd be very uncomfortable telling you "A CCLA is not required by your institution.". Only you can determine whether a CCLA is required in your particular situation.

**Comment by mairi** [comment]
Thanks @apetro for such a comprehensive response.  I have forwarded this information along with some related queries to my line management to double-check.  I'm not aware of any impediment to my contributions being made, but I'm not an expert on intellectual property & employment law, so I can't be 100% clear on the position regarding permission.
